UIKit/UIStatusBarItem.h

```diff
@@ -1,17 +1,62 @@
-typedef NS_ENUM(NSUInteger, UIStatusBarItemType) {
-	UIStatusBarItemTypeDontKnowYetComeBackLater
-};
-
-@class UIStatusBarItemView, UIStatusBarLayoutManager;
+// this enum was reversed out of iOS 10.2, not clear if it's ABI stable
+typedef enum {
+    UIStatusBarItemTypeTime,           // this Time has a centerOrder of 1
+    UIStatusBarItemTypeQuietModeQuietMode,
+    UIStatusBarItemTypeAirplaneModeAirplane,
+    UIStatusBarItemTypeSignalStrength,
+    UIStatusBarItemTypeService,
+    UIStatusBarItemTypePersonName,
+    UIStatusBarItemTypeDataNetwork,
+    UIStatusBarItemTypeDeprecatedTime, // this Time has a centerOrder of 0
+    UIStatusBarItemTypeBattery,
+    UIStatusBarItemTypeBatteryPercent,
+    UIStatusBarItemTypeNotCharging,
+    UIStatusBarItemTypeBluetoothBattery,
+    UIStatusBarItemTypeBluetooth,
+    UIStatusBarItemTypeIndicatorTTY,
+    UIStatusBarItemTypeIndicatorAlarm,
+    UIStatusBarItemTypeIndicatorPlus,
+    UIStatusBarItemTypeIndicator,
+    UIStatusBarItemTypeLocation,
+    UIStatusBarItemTypeIndicatorRotationLock,
+    UIStatusBarItemTypeDoubleHeightDoubleHeight,
+    UIStatusBarItemTypeIndicatorAirPlay,
+    UIStatusBarItemTypeIndicatorSiri,
+    UIStatusBarItemTypeIndicatorCarPlay,
+    UIStatusBarItemTypeIndicatorStudent,
+    UIStatusBarItemTypeIndicatorVPN,
+    UIStatusBarItemTypeIndicatorCallForward,
+    UIStatusBarItemTypeActivity,
+    UIStatusBarItemTypeThermalColor,
+    UIStatusBarItemTypeRadar,
+    UIStatusBarItemTypeElectronicTollCollection,
+    UIStatusBarItemTypeReturnToLastApp,
+    UIStatusBarItemTypeLock,
+    UIStatusBarItemTypeLiquidDetection,
+    UIStatusBarItemTypeIndicatorBTHeadphones,
+    UIStatusBarItemTypeHome,
+    UIStatusBarItemTypeReturnToCall,
+    UIStatusBarItemTypeNavigation,
+    UIStatusBarItemTypeMapsCompass,
+    UIStatusBarItemTypeTethering,
+    UIStatusBarItemTypeBreadcrumb,
+    UIStatusBarItemTypeOpenInSafari
+} UIStatusBarItemType;
 
 @interface UIStatusBarItem : NSObject
 
 @property (nonatomic, readonly) NSString *indicatorName;
+@property (nonatomic, readonly) UIStatusBarItemType type;
+@property (nonatomic, readonly) Class viewClass; // a subclass of UIStatusBarItemView
 
-- (instancetype)initWithType:(UIStatusBarItemType)type;
+@property (nonatomic, readonly) int priority;
+@property (nonatomic, readonly) int leftOrder;
+@property (nonatomic, readonly) int rightOrder;
 
-- (void)setView:(UIStatusBarItemView *)view forManager:(UIStatusBarLayoutManager *)layoutManager;
```

Comment:
I can't find evidence that this ever existed

Reply:
May have been mixed up with some libstatusbar stuff. Agreed, best to remove it.

Reply:
Confirmed: https://github.com/phoenix3200/libstatusbar/blob/dc5877e41891ed57a08593ab2e26fd3e9818bd2b/UIStatusBarCustomItem.mm#L174